var/spack/repos/builtin/packages/llvm-amdgpu/package.py

```diff
@@ -34,6 +34,17 @@ class LlvmAmdgpu(CMakePackage):
     variant('build_type', default='Release', values=("Release", "Debug", "RelWithDebInfo"), description='CMake build type')
     variant('rocm-device-libs', default=True, description='Build ROCm device libs as external LLVM project instead of a standalone spack package.')
     variant('openmp', default=True, description='Enable OpenMP')
+    variant(
+        "llvm_dylib",
+        default=False,
+        description="Build LLVM shared library, containing all "
+        "components in a single shared library",
+    )
+
+    provides('libllvm@11', when='@3.5:3.8.0')
+    provides('libllvm@12', when='@3.9:4.2.0')
+    provides('libllvm@13', when='@4.3:')
+    provides('libllvm +llvm_dylib', when='+llvm_dylib')
```

Comment:
Unfortunately I'm pretty sure the variant here on the `libllvm` virtual has no effect (our fault, but we can fix it in the new concretizer).  Do you need it or is `libllvm` sufficient?  It seems like you're using `libllvm` alone to mean that `llvm_dylib` is enabled on the provider.

Reply:
I thought I would need to map the variants to the virtual package, but if that is done automatically then that is sufficient for this case. The idea here was that `libllvm ~llvm_dylib` was the default and then if `libllvm +llvm_dylib` was requested it would provide `llvm-amdgpu +llvm_dylib`.